File: Tema 06 - React/02. After/petshelter.ui/src/Models/Fundraiser.ts
import type { RealPerson } from './RealPerson';

export class Fundraiser {
  id?:number;
  name: string;
  target: number;
  currentlyRaised?: number;
  dueDate: Date;
  creationDate?: Date;
  status?: FundraiserStatus;
  owner: RealPerson;
  donors?: RealPerson[];

  constructor(
    name: string,
    target: number,
    dueDate: Date,
    owner: RealPerson,
    id?:number,
    creationDate?: Date,
    status?: FundraiserStatus,
    currentlyRaised?: number,
    donors?: RealPerson[],
  ) {
    this.name = name;
    this.target = target;
    this.currentlyRaised = currentlyRaised;
    this.creationDate = creationDate;
    this.dueDate = dueDate;
    this.status = status;
    this.owner = owner;
    this.donors = donors;
    this.id=id;
  }
}

export const enum FundraiserStatus {
  'Closed' = 'Closed',
  'Active' = 'Active',
}

File: Tema 07 - Clean code/After Tema 3/PetShelter.DataAccessLayer/Models/Fundraiser.cs
namespace PetShelter.DataAccessLayer.Models;

public class Fundraiser : IEntity
{
    public int Id { get; set; }
    public string Name { get; set; }
    public decimal CurrentlyRaised { get; set; }
    public decimal Target { get; set; }
    public int OwnerId { get; set; }
    public DateTime CreationDate { get; set; }
    public DateTime DueDate { get; set; }
    public string Status { get; set; }


    public ICollection<Donation> Donations { get; set; }
    public Person Owner { get; set; }
    public Fundraiser(string name, decimal target,int ownerId, DateTime dueDate)
    {
        Name = name;
        Target = target;
        OwnerId = ownerId;
        CreationDate = DateTime.Now;
        DueDate = dueDate;
        Status = "Active";
        CurrentlyRaised = 0;
        Donations= new List<Donation>();
    }
}